### DISCUSSION TITLE
Getting an optional boolean value from a table

### DISCUSSION THREAD

**Discussion by gtokic** [toml-f/toml-f#122] [Q&A] [answered]
I'm trying to read an optional boolean from a TOML table, but I'm not clear what the proper way to set that up is.

The following code returns `iostat = 0` regardless of whether the key `"bool"` is defined in the table or not. If the key is not defined, the variable `bool` returns `.false.`.

```fortran
type(toml_table),allocatable :: table
logical   :: bool
integer   :: iostat

call get_value(table, 'bool', bool, stat=iostat)
```

As I cannot check for `allocated(bool)` or `associated(bool)` as I could for a `character(len=:),allocatable` or a pointer-type variable, what is the way to check whether a key actually exist in a table? Is there a procedure `has_key()` for `toml_table` type or something of that sort that I'm not aware of? 

By the way, the fact that `iostat = 0` for non-existing keys happens also for `integer` and `real` scalar variables.

I'm using TOML-F v0.2.2.

**Comment by awvwgk** [comment, marked as answer]
That is indeed a bug in the getter interface, I'll add a branch to catch this correctly in https://github.com/toml-f/toml-f/pull/123.

**Reply by gtokic** [reply]
I've installed the new v0.4.0 version and it runs great now. Thanks for the quick fix!
